test/Microsoft.AspNet.Mvc.ModelBinding.Test/Metadata/ModelMetadataProviderTest.cs

```diff
@@ -726,6 +726,85 @@ public void Metadata_Null_ForPrivateGetProperty()
             Assert.Null(metadata.Properties[propertyName]);
         }
 
+        [Fact]
+        public void BindingBehavior_GetMetadataForType_UsesDefaultValues()
+        {
+            // Arrange
+            var metadataProvider = TestModelMetadataProvider.CreateDefaultProvider();
+            var type = typeof(BindingBehaviorModel);
+
+            // Act
+            var metadata = metadataProvider.GetMetadataForType(type);
+
+            // Assert
+            Assert.True(metadata.IsBindingAllowed);
+            Assert.False(metadata.IsBindingRequired);
+        }
+
+        [Fact]
+        public void BindingBehavior_Override_RequiredOnProperty()
+        {
+            // Arrange
+            var metadataProvider = TestModelMetadataProvider.CreateDefaultProvider();
+            var type = typeof(BindingBehaviorModel);
+            var propertyName = nameof(BindingBehaviorModel.BindRequiredOverride);
+
+            // Act
+            var metadata = metadataProvider.GetMetadataForProperty(type, propertyName);
+
+            // Assert
+            Assert.True(metadata.IsBindingAllowed);
+            Assert.True(metadata.IsBindingRequired);
+        }
+
+        [Fact]
+        public void BindingBehavior_Override_OptionalOnProperty()
```

Comment:
nit: prefer naming scheme used below to what you're using for these two `_Override_` tests.  for this one how 'bout `BindingBehavior_Optional_SetOnPropertyOverridingClass`?